mypy/binder.py

```diff
@@ -15,7 +15,7 @@
 BindableExpression = Union[IndexExpr, MemberExpr, NameExpr]
 
 
-class Frame(Dict[Key, Type]):
+class Frame(Dict[Optional[Key], Type]):
```

Comment:
I think the actual semantics is that keys should not be ``None``. Therefore it should not be ``Optional`` here. Also, saving few lines of code does not justify adding potentially misleading hints. Strict optional typically touches subtle things, thus everything should be written as "right" as possible.